packages/lodestar/src/util/sleep.ts

```diff
@@ -1,3 +1,30 @@
-export function sleep(ms: number): Promise<void> {
-  return new Promise((resolve) => setTimeout(resolve, ms));
+import {AbortSignal} from "abort-controller";
+import {ErrorAborted} from "./errors";
+
+/**
+ * Abortable sleep function. Cleans everything on all cases preventing leaks
+ * On abort throws ErrorAborted
+ */
+export async function sleep(ms: number, signal?: AbortSignal): Promise<void> {
```

Comment:
Changed the behaviour of sleep on abort to throw an Error. It makes flow control much easier. In the genesis builder it would be as simple to catch in `Chain` for `ErrorAborted` and log or ignore